Source/WebCore/platform/mediastream/gstreamer/GStreamerMediaStreamSource.cpp

```diff
@@ -157,6 +163,23 @@ class InternalSource final : public MediaStreamTrackPrivate::Observer,
             data->m_enoughData = false;
         }), this);
 
+        createGstStream();
+
+        auto pad = adoptGRef(gst_element_get_static_pad(m_src.get(), "src"));
+        gst_pad_add_probe(pad.get(), GST_PAD_PROBE_TYPE_QUERY_UPSTREAM, reinterpret_cast<GstPadProbeCallback>(+[](GstPad*, GstPadProbeInfo* info, InternalSource*) -> GstPadProbeReturn {
+            auto* query = GST_PAD_PROBE_INFO_QUERY(info);
+            switch (GST_QUERY_TYPE(query)) {
+#if GST_CHECK_VERSION(1, 22, 0)
```

Comment:
Considering that we are only reacting on QUERY_SELECTABLE and no-op otherwise, wouldn't it to make more sense flag they whole probe?